wasm/src/main/scala/ir2wasm/HelperFunctions.scala

```diff
@@ -784,4 +787,38 @@ object HelperFunctions {
     fctx.buildAndAddToContext()
   }
 
+  def genCloneInstance(clazz: LinkedClass)(implicit ctx: WasmContext): Unit = {
+    import WasmImmediate._
+    val info = ctx.getClassInfo(clazz.name.name)
+    if (info.interfaces.contains(IRNames.CloneableClass)) {
```

Comment:
```suggestion
    if (info.ancestors.contains(IRNames.CloneableClass)) {
```
Otherwise, indirectly extending `Cloneable` will not be taken into account.

Reply:
https://github.com/tanishiking/scala-wasm/pull/36/commits/15941fc6d2ac33977473948fc1f4ba70bdb3870c